src/aviary/env_client.py

```diff
@@ -134,27 +132,23 @@ def __init__(
         self.server_url = server_url
         self.request_timeout = request_timeout
         self.api_key = api_key
-        self._len: int | object | None = UNSET_LEN
 
-    def get_http_client(self) -> httpx.AsyncClient:
+        with self._get_http_client(httpx.Client) as http_client:
+            response = http_client.get("/info")
+            response.raise_for_status()
+            self._len = cast(int | None, response.json()["dataset_size"])
+
+    def _get_http_client(
+        self,
+        client_class: type[TClient] = httpx.AsyncClient,  # type: ignore[assignment]
```

Comment:
Can we just YAGNI on providing a default value here? This code doesn't use the default anyways